src/Microsoft.Health.Dicom.Client/Models/ExtendedQueryTagStatus.cs

```diff
@@ -0,0 +1,28 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Dicom.Client.Models
+{
+    /// <summary>
+    /// Status of extended query tag.
+    /// </summary>
+    public enum ExtendedQueryTagStatus
+    {
+        /// <summary>
+        /// The query tag is being reindexed.
+        /// </summary>
+        Reindexing = 0,
+
+        /// <summary>
+        /// The query tag has been added to system.
+        /// </summary>
+        Added = 1,
+
+        /// <summary>
+        /// The query tag is being deindexed.
+        /// </summary>
+        Deindexing = 2,
```

Comment:
On 2nd though, is this exposing too much of our internals. Should this just be like `Adding`, `Ready`, and Removing`?

Reply:
will fix it.

Reply:
may be deleting instead of Removing to keep consistency with API name DeleteExtendedQueryTag?